lib/src/error_tracking/dart_exception_processor.dart

```diff
@@ -0,0 +1,335 @@
+import 'package:stack_trace/stack_trace.dart';
+import 'utils/isolate_utils.dart' as isolate_utils;
+
+class DartExceptionProcessor {
+  /// Converts Dart error/exception and stack trace to PostHog exception format
+  static Map<String, dynamic> processException({
+    required Object error,
+    StackTrace? stackTrace,
+    Map<String, Object>? properties,
+    bool handled = true,
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+    StackTrace Function()? stackTraceProvider, //for testing
+  }) {
+    StackTrace? effectiveStackTrace = stackTrace;
+    bool isGeneratedStackTrace = false;
+
+    // If it's an Error, try to use its built-in stackTrace
+    if (error is Error) {
+      effectiveStackTrace ??= error.stackTrace;
+    }
+
+    // If still null or empty, get current stack trace
+    if (effectiveStackTrace == null ||
+        effectiveStackTrace == StackTrace.empty) {
+      effectiveStackTrace = stackTraceProvider?.call() ?? StackTrace.current;
```

Comment:
you have to check again for StackTrace.empty btw since `StackTrace.current` could return empty